### PR TITLE
make smart quoting available as a flag

### PR DESCRIPTION
**Pull request by maguec**
Handle the case where there are quotes around an entry that cause the following error
```
Traceback (most recent call last):                                                                                                                                                     
  File "bulk_insert.py", line 399, in <module>                                                                                                                                         
    bulk_insert()                                                                                                                                                                      
  File "/home/chris/.local/lib/python3.6/site-packages/click/core.py", line 764, in __call__                                                                                           
    return self.main(*args, **kwargs)                                                                                                                                                  
  File "/home/chris/.local/lib/python3.6/site-packages/click/core.py", line 717, in main                                                                                               
    rv = self.invoke(ctx)                                                                                                                                                              
  File "/home/chris/.local/lib/python3.6/site-packages/click/core.py", line 956, in invoke                                                                                             
    return ctx.invoke(self.callback, **ctx.params)                                                                                                                                     
  File "/home/chris/.local/lib/python3.6/site-packages/click/core.py", line 555, in invoke                                                                                             
    return callback(*args, **kwargs)                                                                                                                                                   
  File "bulk_insert.py", line 387, in bulk_insert                                                                                                                                      
    process_entity_csvs(Label, nodes)                                                                                                                                                  
  File "bulk_insert.py", line 307, in process_entity_csvs                                                                                                                              
    entity = cls(in_csv)                                                                                                                                                               
  File "bulk_insert.py", line 169, in __init__                                                                                                                                         
    self.process_entities(expected_col_count)                                                                                                                                          
  File "bulk_insert.py", line 192, in process_entities                                                                                                                                 
    self.validate_row(expected_col_count, row)                                                                                                                                         
  File "bulk_insert.py", line 133, in validate_row                                                                                                                                     
    % (self.infile.name, self.reader.line_num, expected_col_count, len(row), ','.join(row)))                                                                                           
__main__.CSVError: Movies.csv:49 Expected 2 columns, encountered 3 ('tt0000049,"Boxing Match; or,Glove Contest"')  ```

### CODE REVIEW

File: bulk_insert.py [hunk 1]
```diff
@@ -104,7 +104,8 @@ def __init__(self, filename):
         self.infile = io.open(filename, 'rt')
         # Initialize CSV reader that ignores leading whitespace in each field
         # and does not modify input quote characters
-        self.reader = csv.reader(self.infile, skipinitialspace=True, quoting=csv.QUOTE_NONE)
+        #self.reader = csv.reader(self.infile, skipinitialspace=True)
```
Comment (jeffreylovitz) [inline bulk_insert.py]: Delete commented line

File: bulk_insert.py [hunk 2]
```diff
@@ -328,16 +329,24 @@ def process_entity_csvs(cls, csvs):
 @click.option('--max-token-count', '-c', default=1024, help='max number of processed CSVs to send per query (default 1024)')
 @click.option('--max-buffer-size', '-b', default=2048, help='max buffer size in megabytes (default 2048)')
 @click.option('--max-token-size', '-t', default=500, help='max size of each token in megabytes (default 500, max 512)')
+@click.option('--max-token-size', '-t', default=500, help='max size of each token in megabytes (default 500, max 512)')
```
Comment (jeffreylovitz) [inline bulk_insert.py]: Duplicate option line

File: bulk_insert.py [hunk 3]
```diff
@@ -328,16 +329,24 @@ def process_entity_csvs(cls, csvs):
...
+@click.option('--quote-minimal/--no-quote-minimal', '-q/-d', default=False, help='only quote those fields which contain special characters such as delimiter, quotechar or any of the characters in lineterminator')
```
Comment (jeffreylovitz) [inline bulk_insert.py]: I don't think the `-d` alternate is strictly necessary, but if you think it is helpful then I have no objection!

Comment (maguec) [reply bulk_insert.py]: just kind of helpful for me as I'm lazy

**Comment by maguec** [comment]
@jeffreylovitz removed the duplicate line and the commented line.  pls review

**Comment by jeffreylovitz** [comment]
Thanks, @maguec!

**Pull request merged**